api/xds_protocol.rst

```diff
@@ -613,6 +613,35 @@ adding/removing/updating clusters. On the other hand, routes are not
 warmed, i.e., the management plane must ensure that clusters referenced
 by a route are in place, before pushing the updates for a route.
 
+.. _xds_protocol_TTL:
+
+TTL
+~~~
+
+In the event that the management server becomes unreachable, the last known configuration received
+by Envoy will persist until the connection is reestablished. For some services, this may not be
+desirable. For example, in the case of a fault injection service, a management server crash at the
+wrong time may leave Envoy in an undesirable state. The TTL setting allows Envoy to remove a set of
+resources after a specified period of time if contact with the management server is lost. This can
+be used, for example, to terminate a fault injection test when the management server can no longer
+be reached.
+
+For clients that support the 'envoy.config.ttl' client feature, A TTL field may be specified on 
```

Comment:
Can you add this to: https://www.envoyproxy.io/docs/envoy/latest/api/client_features? Also, I would recommend naming this feature something like 'envoy.config.supports_resource_in_sotw_response' or something like that since we may make use of this for other resource things in the future? @htuch @markdroth wdyt?

Reply:
I agree, we should definitely have a "envoy.config.supports_resource_in_sotw_response" client feature to specifically control sending `Resource` messages in SotW responses, since we'll want this for the naming work as well.  And we should document that the SotW protocol does not support TTLs without supporting this.

Do we also want a separate client feature for supporting TTLs?  The advantage of that might be that a management server might decide that it's too dangerous to send a fault injection config to a client that does not support TTLs.  Alternatively, we could skip having a feature for this and just force management servers to accept that some clients may ignore the TTLs.  I don't know enough about the use-cases to know which one is better.

Reply:
> I don't know enough about the use-cases to know which one is better.

I could go either way, but my gut would say that we should have an additional client feature for TTL also. @htuch wdyt?